assets/src/components/shuttleMapPage.tsx

```diff
@@ -57,39 +74,160 @@ const ShuttleMapPage = (): ReactElement<HTMLDivElement> => {
+      />
+    </div>
+  )
+}
+
+interface ShuttleMapProps {
+  // TODO: revisit this prop
+  selectedShuttleRunIds: RunId[] | "all"
+  //
+  selectedVehicleId?: VehicleId
+  selectedShuttles: Vehicle[]
+  selectedRouteShapes?: Shape[]
+  trainVehicles?: TrainVehicle[]
+  tileType?: TileType
+  setTileType: (tileType: TileType) => void
+  selectVehicle: (vehicle: Vehicle | Ghost) => void
+}
+
+export const ShuttleMap = ({
+  selectedShuttles,
+  selectedShuttleRunIds,
+  selectedVehicleId,
+  selectedRouteShapes,
+  trainVehicles,
+  selectVehicle,
+  tileType,
+  setTileType,
+}: ShuttleMapProps) => {
+  const positions = selectedShuttles.map(vehicleToLeafletLatLng)
+
+  type FollowerController = false | "user-location" | "vehicle-location"
+  const [followerController, setFollowerController] =
+    useState<FollowerController>("vehicle-location")
+
+  const state = useInteractiveFollowerState(),
+    { setShouldFollow: setFollowActive } = state
 
-      <div className="c-shuttle-map__map">
-        <MapFollowingSelectionKey
-          selectedVehicleId={selectedVehicleOrGhost?.id}
-          vehicles={selectedShuttles}
-          shapes={shapes}
-          trainVehicles={trainVehicles}
-          onPrimaryVehicleSelect={(vehicle) => dispatch(selectVehicle(vehicle))}
-          selectionKey={followerResetKey}
-          tileType={tileType}
-        >
-          <LayersControl.WithTileContext
-            setTileType={(tileType: TileType) =>
-              dispatch(setTileType("shuttleMap", tileType))
-            }
-          />
-        </MapFollowingSelectionKey>
-      </div>
+  const setShouldFollow = (controller: FollowerController) => {
+    setFollowerController(controller)
+    setFollowActive(controller !== false ? true : false)
+  }
+
+  const [location, setLocation] = useState<null | GeolocationCoordinates>(null)
+
+  const followPositions = useMemo(
+    () =>
+      followerController
+        ? {
+            "user-location": location
+              ? [latLng(location.latitude, location.longitude)]
+              : [],
+            "vehicle-location": positions,
+          }[followerController]
+        : [],
+    [followerController, location, positions]
+  )
+
+  useEffect(() => {
+    if (followerController === false) {
+      setShouldFollow("vehicle-location")
+    }
+    // This effect is an "event", and not "reactive" to all it's dependencies.
+    // Because we _only_ want to run this when the run id's change, we need to
+    // suppress the exhaustive-deps lint until `useEffectEvent` is stabilized
+    //
+    // See the react docs:
+    // > ## Is it okay to suppress the dependency linter instead?
+    // > After `useEffectEvent` becomes a stable part of React,
+    // > we recommend never suppressing the linter.
+    // > -- https://react.dev/learn/separating-events-from-effects#is-it-okay-to-suppress-the-dependency-linter-instead
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedShuttleRunIds])
```

Comment:
**suggestion:** would it be possible to add `followerController` and `setShouldFollow` as dependencies, but further gate the `setShouldFollow` call behind some logic that verifies that `selectedShuttleRunIds` actually changed? We do sort of similar things in some places where we care about deep equality rather than referential equality in dependencies, like in the `useSwings` hook for instance, where there's an additional check to see if the dependencies actually changed in the specific manner that we care about before performing the logic.

Reply:
`useMinischeduleRuns` might actually be a better example.

Reply:
I'll look into it. I avoided it because I didn't want to introduce more state to manage, especially state that is essentially just noise to the actual logic of the component, and I didn't want to wrap up that noise in an abstraction too soon. I'm personally a fan of this approch, because it makes it easy to upgrade to using the `useEffectEvent` api and I believe this to be the "correct" way to handle this. But I'll see about reimplementing it.

Reply:
I wish it was possible to pass parameters to the linker to have it selectively ignore rules, so for instance in this case we could tell it to ignore `followerController` and `setShouldFollow` but if you introduce any new inputs in the body without adding them to the dependencies array it will catch them.

Reply:
Yeah that would certainly be better 😞

Reply:
d9f3abe79a10c26d61a59528064c7c0504ea7e20